src/constants.ts

```diff
@@ -2,27 +2,125 @@ import { Endpoints } from './models/core/Endpoints';
 
 export const defaultApiVersion = 20220511;
 
-export const defaultEndpoints: Required<Endpoints> = {
-  universalSearch: 'https://liveapi.yext.com/v2/accounts/me/search/query',
-  verticalSearch: 'https://liveapi.yext.com/v2/accounts/me/search/vertical/query',
-  questionSubmission: 'https://liveapi.yext.com/v2/accounts/me/createQuestion',
-  status: 'https://answersstatus.pagescdn.com',
-  universalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/search/autocomplete',
-  verticalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/search/vertical/autocomplete',
-  filterSearch: 'https://liveapi-cached.yext.com/v2/accounts/me/search/filtersearch',
-} as const;
+export enum CloudRegion {
+  US = 'us',
+  EU = 'eu',
+}
+
+export enum Environment {
+  PROD = 'prod',
+  SANDBOX = 'sbx',
+}
 
 /**
- * The endpoints to use for sandbox experiences.
+ * Provides methods for getting various endpoints.
  *
  * @public
  */
-export const SandboxEndpoints: Required<Endpoints> = {
-  universalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/query',
-  verticalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/vertical/query',
-  questionSubmission: 'https://liveapi-sandbox.yext.com/v2/accounts/me/createQuestion',
-  status: 'https://answersstatus.pagescdn.com',
-  universalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/autocomplete',
-  verticalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/vertical/autocomplete',
-  filterSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/filtersearch',
-} as const;
\ No newline at end of file
+export class EndpointsProvider {
+  private readonly environment: Environment;
+  private readonly cloudRegion: CloudRegion;
+  private readonly overrideEndpoints?: Endpoints;
+
+  constructor(environment?: Environment, cloudRegion?: CloudRegion, overrideEndpoints?: Endpoints) {
+    this.environment = environment || Environment.PROD;
+    this.cloudRegion = cloudRegion || CloudRegion.US;
+    this.overrideEndpoints = overrideEndpoints;
+  }
+
+  /** @internal */
+  getDomain() {
+    return `https://${this.environment}-cdn.${this.cloudRegion}.yextapis.com`;
```

Comment:
which domain is this? i don't think it is the right one for us live API?

Reply:
It's the new naming convention: https://docs.google.com/document/d/1pu4IhzubaChPPMX5Vmu3LeXjOfyIR6cWBx2YfNizi_E/edit#